test/index.js

```diff
@@ -3012,12 +3022,14 @@ See https://github.com/sanctuary-js/sanctuary-type-classes/tree/v${Z$version}#Al
     const map =
     def ('map')
         ({f: [Z.Functor]})
-        ([$.Function ([a, b]), f (a), f (b)])
+        ([$.Fn (a) (b), f (a), f (b)])
         (curry2 (Z.map));
 
     eq (map (Math.sqrt) (Nothing)) (Nothing);
     eq (map (Math.sqrt) (Just (9))) (Just (3));
 
+    eq (map (Math.sqrt) (Math.sqrt) (625)) (5);
```

Comment:
The expression above type-checks because this pull request adds `Fn ??? ???` to the default environment.